shared/test/middleware/helpers/test_asset_bucket.rb

```diff
@@ -0,0 +1,86 @@
+require_relative '../../files_api_test_base' # Must be required first to establish load paths
+require_relative '../../files_api_test_helper'
+require 'ostruct'
+
+#
+# Unlike AssetsTest (test_assets.rb) which is essentially a controller test
+# designed to full exercise the public API, this is meant to be more like a
+# unit test for AssetBucket (asset_bucket.rb).
```

Comment:
it sure would have been nice to build our project infrastructure all on rails! 😛